### PR TITLE
Add embeddable code sample runner via carcin-play

### PR DESCRIPTION
**Pull request by straight-shoota**
Adds [carcin-play](https://straight-shoota.github.io/carcin-play/), an embeddable playground which allows to run code samples directly inline. The code can be edited and executed (via https://carc.in).
This is specifically useful for the upcoming tutorials because it allows the reader to follow the lessons and play with the code. But it could be used with existing code samples as well.

Usage: 

````md
```{.crystal, .crystal-play}
puts "hello world"
```
```` 

grafik

I'm not very proud of the makeshift dependency management via makefile, but it seems like the simplest solution to get started. 
For now the dependencies are just pulled into the repository. Update via `make clean_vendor vendor`. It does the job. We can evaluate using a proper tool for this later.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -27,6 +27,35 @@ $(MKDOCS): $(PIP) requirements.txt
...
+.PHONY: vendor
+vendor: ## Install assets from external sources
+vendor: carcin-play codemirror ansi_up
+
+.PHONY: carcin-play
+carcin-play: docs/assets/vendor/carcin-play/carcin.js docs/assets/vendor/carcin-play/carcin-play.js docs/assets/vendor/carcin-play/carcin-play.css docs/assets/vendor/carcin-play/codemirror-theme.css
+
+docs/assets/vendor/carcin-play/%:
+	mkdir -p $(@D)
+	wget -O $@ https://github.com/straight-shoota/carcin-play/raw/master/$(@F)
```
[the remark this answers is not in the document]
Comment (straight-shoota) [reply Makefile:39]: Sure

**Comment by straight-shoota** [comment]
@sdogruyol I don't have a preview in this repo yet, but this basically catches it: https://straight-shoota.github.io/carcin-play/samples/hello-world/ 
I'm going to post a preview of the first tutorial lessons later.

**Review by Fryguy — APPROVED** [review]

**Comment by bcardiff** [comment]
I would use yarn for managing this dependencies.

I am worried if carc.in can be used for this. The load on it would be much higher I expect. I think @jhass should have a word here.

**Comment by Fryguy** [comment]
@bcardiff Should crystal itself host this on something like a play.crystal-lang.com?

**Comment by straight-shoota** [comment]
carc.in is already aliased as https://play.crystal-lang.org/

**Comment by jhass** [comment]
Right now this runs on fairly low resources (1 CPU, 1GB RAM, 4GB swap) and I could commit some more resources to it but at some point it would certainly be great to move this to project resources.

My bigger concern is that I currently don't feel like I have the time and motivation to give it the rewrite it needs and keeping the current solution running is something of a hairy ball sometimes to be honest.

So we can try this out but I'd like a back-out plan :)

**Comment by straight-shoota** [comment]
Yes, long term we're gonna need to improve on that. But I think the current state of carc.in should be fine for now. It has proven to work pretty reliably, despite its issues. Of course we'll have to invest in improving the implementation in the forseeable future. But it's a solid start and available now.

I was actually surprised about the low VM footprint, but I guess it doesn't see much concurrent action. But this also means there's much space to grow for little effort (just add a few more cores and memory for small money).

And I don't think there's any reasonable alternative. Maybe we could cooperate with some other online playground provider like repl.it. But I suppose they would like people to use their platforms and are usually lagging behind a couple of releases.
I think it has great value to host our own Crystal playground. We can start building a good one (= improved rewrite of carc.in), but it's possible to delay that and keep going with the current carc.in for now.

**Comment by Fryguy** [comment]
Looking at the tutorial PR, this is really cool.  It makes a huge difference in initial accessibility for someone coming to learn, as they don't need to install anything and can just start immediately.

**Pull request merged**